### ISSUE TITLE
`linebreaks` never respells a line break another rule writes in the same run

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#352]
Stylelint runs the rules of a plugin in the order the configuration spells them, and it runs each of them once. A line break written by one rule is therefore invisible to `linebreaks` whenever `linebreaks` is listed first, and the file comes out broken two ways at once.

## The run

Fixture `a{color:red}\n`, both rules at once, only the order changed:

```
block-closing-brace-newline-before first -> "a{color:red\r\n}\r\n"   nothing left to report
linebreaks first                         -> "a{color:red\n}\r\n"     one warning left
```

With `linebreaks: "windows"` listed first, the closing brace is moved onto a line of its own behind a line feed, and that line feed is never respelled. The file now holds both spellings. A second run of `--fix` repairs it, and the first run reports nothing at all.

## Why

`lintPostcssResult.mjs` of `stylelint@17.14.1` sorts the rules of a run by their place in Stylelint's own registry, and a plugin's rules are not in it:

```js
const rulesOrder = new Map(Object.keys(rules).map((ruleName, index) => [ruleName, index]))

ruleNames = Object.keys(config.rules).sort((a, b) => (rulesOrder.get(a) ?? -1) - (rulesOrder.get(b) ?? -1))
```

Every rule of this plugin resolves to `-1`, the sort is stable, and so the order is the one the user's configuration happens to spell. `lintPostcssResult` is called once per lint; there is no loop to a fixed point.

## The counterparts

Nine rules write a line break that `linebreaks` may never see. Every one of them was measured to answer differently under the two orders:

- `block-closing-brace-newline-before`
- `block-opening-brace-newline-after`
- `declaration-block-semicolon-newline-after`
- `declaration-block-semicolon-newline-before`
- `function-comma-newline-after`
- `function-comma-newline-before`
- `function-parentheses-newline-inside`
- `selector-list-comma-newline-after`
- `selector-list-comma-newline-before`

## What ends it

A rule that writes a line break has to write the break the configuration asks for, rather than a line feed for `linebreaks` to correct afterwards. That means reading the setting of `linebreaks` out of `result.stylelint.config` — which holds every rule's normalised settings and is assigned before any rule runs — through a shared helper, so that the answer is the same wherever a break is written.

The measurement ran 79 rules under 208 primary options over eight fixtures of plain CSS.

One of the four classes of [#356](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/356), which is about the class rather than about any member of it.
